### DISCUSSION TITLE
delay of receiving trace

### DISCUSSION THREAD

**Discussion by zimosi** [apache/skywalking#12427] [General]
I have to wait about 20 seconds to see the trace on the UI after making a request. Is there any way to reduce the waiting time of seeing trace to less than five seconds

**Comment by wu-sheng** [comment]
If the request is quick, then most likely, your database is responsible for the latency, or OAP has limited CPU resources.

OAP set up metrics latency for 20s to reduce database payload, but not for the trace.

**Reply by zimosi** [reply]
Thank you for your reply, just for clearification. Do you mean that the metrics should have 20s latency, but trace and log information should be synchronous? Also, I connected ElasticSearch database to OAP.  If this is the reason, I will check the latency of the database.

**Reply by wu-sheng** [reply]
Trace will have latency for sure, but not much. Async process takes some time.

**Reply by zimosi** [reply]
Got it, then what is the normal range of Trace latency?

**Reply by wu-sheng** [reply]
several seconds. I think.
